rust/src/server.rs

```diff
@@ -159,8 +164,14 @@ fn initialize(
         ..
     } = config;
 
-    fs::create_dir_all(blocks_dir.clone()).ok();
-    fs::create_dir_all(staking_ledgers_dir.clone()).ok();
+    if let Err(e) = fs::create_dir_all(&blocks_dir) {
+        error!("Failed to create blocks directory in {blocks_dir:#?}: {e}");
+        process::exit(1);
+    }
+    if let Err(e) = fs::create_dir_all(&staking_ledgers_dir) {
+        error!("Failed to create staking ledgers directory in {staking_ledgers_dir:#?}: {e}");
+        process::exit(1);
```

Comment:
Should we use different exit codes to signify something?

Reply:
I suspect that's a bigger conversation